### DISCUSSION TITLE
Add GraphQL Support

### DISCUSSION THREAD

**Discussion by Intevel** [Intevel/nuxt-directus#46] [General]

**Comment by productdevbook** [comment]
Why was this topic closed ?

**Comment by vanling** [comment]
> Why was this topic closed ? 

I guess because it has been added to the in v1.8.0.

**Comment by productdevbook** [comment]
https://github.com/Intevel/nuxt-directus/commit/358bbf6d4c418741c5528b3d083f054d6efbb355

Yes, I saw the code now, but I think it might be a better structure, I will work on it.

**Comment by productdevbook** [comment]
```Vue Apollo``` -> https://v4.apollo.vuejs.org/

I'm thinking of integrating that into this build. I did it in Vue, but I don't know how to do it on nuxt side, I have to research it. Maybe it will help when I create a pr.

But I don't know if it is right to integrate this structure here.

**Comment by Intevel** [comment]
Any updates on this? @productdevbook

**Comment by productdevbook** [comment]
Since I have little free time per week, I am in research.  When I find a logical solution, I will give information here.  I am working on monorepo in a similar project.  I haven't been able to find sharp solution yet.

**Comment by narduin** [comment]
> [358bbf6](https://github.com/Intevel/nuxt-directus/commit/358bbf6d4c418741c5528b3d083f054d6efbb355)
> 
> Yes, I saw the code now, but I think it might be a better structure, I will work on it.

Hi!
Is it currently possible to use graphql requests with this module? I'm not quite sure of how to proceed, sorry…

**Comment by Intevel** [comment]
Currently not.

**Comment by productdevbook** [comment]
The question is, how does it make sense to include it here when there is apollo graphql?

**Reply by productdevbook** [reply]
Maybe we can just present in this library, bring the schema.graphql download feature to the main directory. So we use schema.graphql in apollo.

**Reply by Intevel** [reply]
So Directus has GraphQL Endpoints, and I think it would be nice to integrate them in our module. But I think it will be a lot of work and I dont know if its so much "needed"

**Reply by productdevbook** [reply]
I think it's not necessary, there is vue-apollo and graphql codegen.

**Reply by productdevbook** [reply]
It would be helpful to just bring the schema download feature.

**Reply by productdevbook** [reply]
If you bring this feature I will create a sample template. People can easily use github templates.

**Reply by nsakibp** [reply]
Is there any update on this, please?

**Reply by narduin** [reply]
I don't know yet about nuxt 3 + apollo but with nuxt 2, there were some cases where apollo was just not working properly ([see this issue](https://github.com/nuxt-community/apollo-module/issues/339)).
I moved away from it in favour of [another module](https://github.com/Gomah/nuxt-graphql-request/) but this one does not support nuxt 3 as of today.

So personally I would love to see graphql integrated here :)

**Comment by nsakibp** [comment]
Thanks a lot for your prompt response
Do you have any working GraphQL sample on Nuxt 2, with the module you
mentioned?

Thanks a lot again!

**Reply by narduin** [reply]
I don't have an example to show right now but I have made [this article](https://www.nardu.in/en/snippets/nuxt-graphql-static/) explaining how to use it :)

**Comment by Corentin7301** [comment]
Hi! 👋🏻
I'm bouncing off the discussion about GraphQL support in the `nuxt-directus` module.
I think it would be a good thing to integrate it into the module.
I tried to use, with Nuxt 3, the `nuxt-graphql-client` module, everything works fine until I ask for no-public data (available only for a User account for example).
From that moment on, I can't retrieve this data via GraphQL because authentication is managed via `nuxt-directus` but `nuxt-graphql-client` doesn't benefit from user management.
Either I don't know how to do this, in which case please help me 😉 , or I think it would be a good thing to include this in the module.
Thanks and have a nice day! 🤘🏻
